crates/wasi-nn/src/registry/mod.rs

```diff
@@ -0,0 +1,16 @@
+//! Define the registry API.
+//!
+//! A [`GraphRegistry`] is place to store backend graphs so they can be loaded
+//! by name. This API does not mandate how a graph is loaded or how it must be
+//! stored--it could be stored remotely and rematerialized when needed, e.g. A
+//! naive in-memory implementation, [`InMemoryRegistry`] is provided for use
+//! with the Wasmtime CLI.
+
+mod in_memory;
+
+use crate::Graph;
+pub use in_memory::InMemoryRegistry;
+
+pub trait GraphRegistry: Send + Sync {
+    fn get_mut(&mut self, name: &str) -> Option<&mut Graph>;
```

Comment:
Another thing I'm not quite sure of is whether this should return `Option<&mut Graph>`, `Option<&Graph>`, or even `Option<Graph>` (seeing as how `Graph` is just an `Arc`-wrapper). We immediately clone the graph and I removed the mutability requirement for `BackendGraph::init_execution_context`... @geekbeast , any opinions from implementing other kinds of `GraphRegistry`?